### PR TITLE
Add error handling tests

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/sequential/sequencerStaking.rules.test.ts [hunk 1]
```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
...
+    const api = getApi();
+    const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationMembers[0],
+        Maintenance.switchMaintenanceModeOff(),
+      ),
+    );
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("false");
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:410]: perhaps you can move this to the beforeEach?

File: test/sequential/sequencerStaking.rules.test.ts [hunk 2]
```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
...
+    expect(resolutionErr).toEqual("WrongCancelRequestId");
+    maintenanceStatus = await api.query.maintenance.maintenanceStatus();
+    expect(maintenanceStatus.isMaintenance.toString()).toEqual("true");
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(
+        foundationMembers[0],
+        Maintenance.switchMaintenanceModeOff(),
+      ),
+    );
+  });
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:480]: I think u forgot to check the status of the deposit, did it work ? Can u ensure if it did or did not? 
Why reusing the same txIndex for both deposit and cancel resolution?

File: test/sequential/sequencerStaking.rules.test.ts [hunk 3]
```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
...
+      )
+      .withDeposit(
+        txIndex + 1,
+        sequencer.keyRingPair.address,
+        sequencer.keyRingPair.address,
+        BN_MILLION,
+      )
+      .on(chain)
+      .buildUnsafe();
+    await waitForSessionChange();
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:522]: why waiting for a new session? waiting for readrights must be enough

File: test/sequential/sequencerStaking.rules.test.ts [hunk 4]
```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
...
+    await waitNewBlock();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
+    );
+    const event2: any = await waitForEvents(api, "tokens.Endowed", 40);
+    const eventData = JSON.parse(
+      JSON.stringify(event2[0]!.event.toHuman().data),
+    );
+    const currencyId = new BN(eventData.currencyId.replaceAll(",", ""));
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:532]: use stringToBN plz

File: test/sequential/sequencerStaking.rules.test.ts [hunk 5]
```diff
@@ -381,6 +393,150 @@ describe("sequencerStaking", () => {
...
+      )
+      .on(chain)
+      .buildUnsafe();
+    await waitForSessionChange();
+    await waitNewBlock();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
+    );
+    const event2: any = await waitForEvents(api, "tokens.Endowed", 40);
```
Comment (goncer) [inline test/sequential/sequencerStaking.rules.test.ts:528]: There is a waiter for the update to be executed, please take a look and ping me if you do not find it.

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/sequencerStaking.rules.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.rules.test.ts:410]: There are other tests in this file. I can create describe fro these three tests or create a new file, I don't know which variant is the best

File: test/sequential/sequencerStaking.rules.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.rules.test.ts:522]: Removed

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 6]
```diff
@@ -4,19 +4,82 @@
...
+  expectMGAExtrinsicSuDidSuccess,
+  filterAndStringifyFirstEvent,
+  waitForEvents,
+  waitNewBlock,
+} from "../../utils/eventListeners";
+import { BN } from "@polkadot/util";
+import { AssetWallet, User } from "../../utils/User";
+import { getAssetIdFromErc20 } from "../../utils/rollup/ethUtils";
+
+async function checkAndSwitchMnModeOff() {
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:44]: Typo! MmModeOff, 
all is about MaintenaceMode :) not Mn

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 7]
```diff
@@ -456,3 +519,135 @@ describe.skip("updateL1FromL1 - errors", () => {
...
+
+  beforeAll(async () => {
+    await initApi();
+    setupUsers();
+    await setupApi();
+    api = getApi();
+  });
+
+  beforeEach(async () => {
+    //TODO: Replace this by some monitoring of the active queue.
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:537]: remove it, careful when copy paste plz

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 8]
```diff
@@ -456,3 +519,135 @@ describe.skip("updateL1FromL1 - errors", () => {
...
+    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:538]: why waiting for DP?

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 9]
```diff
@@ -456,3 +519,135 @@ describe.skip("updateL1FromL1 - errors", () => {
...
+    await SequencerStaking.removeAddedSequencers(10);
+    chain = "Ethereum";
+    sequencer = await setupASequencer(chain);
+    txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+  });
+
+  it("When a cancel resolution fail, maintenance mode will be triggered automatically", async () => {
+    await checkAndSwitchMnModeOff();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:547]: we must create a constant for those waiting times ( 50 ) and the 40 below, the day we have some changes on the dp, sequencer selection we  will cry,

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 10]
```diff
@@ -456,3 +519,135 @@ describe.skip("updateL1FromL1 - errors", () => {
...
+      )
+      .on(chain)
+      .buildUnsafe();
+    await waitNewBlock();
+    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
+    const event2 = await Sudo.batchAsSudoFinalized(
+      Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
+    );
+    const eventFiltered = filterAndStringifyFirstEvent(event2, "L1ReadStored");
+    await Rolldown.waitForL2UpdateExecuted(stringToBN(eventFiltered.range.end));
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:642]: plz validate that the overflow error event is not in the list of events

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 11]
```diff
@@ -456,3 +519,135 @@ describe.skip("updateL1FromL1 - errors", () => {
     );
   });
 });
+
+describe("updateL1FromL1 - cancelResolution and deposit errors", () => {
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:523]: Changed

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 8 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:538]: Doesn't need, deleted

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 7 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:537]: Deleted

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:44]: There was MmModeOff here, but I renamed it to MmOff

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 9 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:547]: Fixed here, and I can create a fast PR where I'll change it for all our tests

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 10 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:642]: Done, but I changed a function

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 12]
```diff
@@ -534,17 +512,21 @@ describe("updateL1FromL1 - cancelResolution and deposit errors", () => {
...
   beforeEach(async () => {
-    //TODO: Replace this by some monitoring of the active queue.
-    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
     await SequencerStaking.removeAddedSequencers(10);
     chain = "Ethereum";
-    sequencer = await setupASequencer(chain);
+    [sequencer] = await setupUsers();
+    await SequencerStaking.setupASequencer(sequencer, chain);
     txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+    waitingPeriod = SequencerStaking.blocksForSequencerUpdate() * 5;
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:520]: what is that magic number * 5 ?

File: test/sequential/rolldown.updateL1FromL1.test.ts [hunk 13]
```diff
@@ -634,12 +624,21 @@ describe("updateL1FromL1 - cancelResolution and deposit errors", () => {
...
+    );
+    await Sudo.batchAsSudoFinalized(
       Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
     );
-    const eventFiltered = filterAndStringifyFirstEvent(event2, "L1ReadStored");
-    await Rolldown.waitForL2UpdateExecuted(stringToBN(eventFiltered.range.end));
+    const event2 = await waitForEvents(
+      api,
+      "rolldown.RequestProcessedOnL2",
+      (await Rolldown.disputePeriodLength()).toNumber() * 4,
```
Comment (goncer) [inline test/sequential/rolldown.updateL1FromL1.test.ts:638]: what is that *4 about?

**Review by goncer — APPROVED** [review]

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 12 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:520]: We needed to replace the number 50 with a number tied to some constant, I chose blocksForSequencerUpdate

File: test/sequential/rolldown.updateL1FromL1.test.ts (hunk 13 above)
Comment (AlexChetverov) [reply test/sequential/rolldown.updateL1FromL1.test.ts:638]: We needed to replace the number 40 with a number tied to some constant, I chose disputePeriodLength

**Pull request merged**